sdk/maps/azure-maps-search/src/main/java/com/azure/maps/search/models/BaseSearchOptions.java

```diff
@@ -3,24 +3,29 @@
 
 package com.azure.maps.search.models;
 
-import java.util.List;
 import java.util.Optional;
 
 import com.azure.core.models.GeoBoundingBox;
+import com.azure.core.models.GeoPosition;
 
 /**
  * Class holding optional parameters for Search.
  *
- * @param <T> The extending type.
  */
-public abstract class BaseSearchOptions<T extends BaseSearchOptions<T>> {
+
+public  class BaseSearchOptions {
     private Integer top;
-    private Integer skip;
-    private List<String> countryFilter;
-    private Integer radiusInMeters;
+    private String query;
+    private String addressLine;
+    private String countryRegion;
     private GeoBoundingBox boundingBox;
-    private String language;
-    private LocalizedMapView localizedMapView;
+    private String view;
+    private GeoPosition coordinates;
+    private String adminDistrict;
+    private String adminDistrict2;
+    private String adminDistrict3;
+    private String locality;
+    private String postalCode;
```

Comment:
This is me not knowing Maps Search to a good extent, but looking at some of the generated enums I saw `adminDistrict` and `adminDistrict2` but not `adminDistrict3` and also saw `postalCode`, `postalCode2`, `postalCode3`, and `postalCode4`, which there is only `postalCode` here. Could you include implementation documentation on why that is as just looking at the APIs here I don't understand why that is.

Reply:
They're for different API. BaseSearchOption.java is for forward geocode API, it doesn't support postal_code 2,3,4 as input parameters; while BoundaryResultTypeEnum.java is the response for search polygon API and it can return postal_code 2,3,4.